picoquic/picoquic.h

```diff
@@ -1056,18 +1056,38 @@ int picoquic_mark_datagram_ready(picoquic_cnx_t* cnx, int is_ready);
  * number of bytes at the provided address, and provide a return code 0 from
  * the callback in case of success, or non zero in case of error.
  * 
- * There may be case when the application marked the context ready for
- * datagrams, but cannot provide data when polled, for example because the
- * length of memory announced in the callback is not long enough for
- * the next application datagram. In that case, if the application either
- * doesn't request a buffer or requests a buffer of length 0, the application
- * will be called again when the next packet is ready to be sent. If the
- * application realizes that it has no more data to send, it MUST call
+ * There are two variants of "picoquic_callback_prepare_datagram", the old one
+ * and the new one, ""picoquic_callback_prepare_datagram_ex", which adds
+ * an "is_active" parameter. This parameter helps handling some cases:
+ * 
+ * - if the application marked the context ready by mistake, it 
+ *   should set the "length" argument to 0, and the "is_active" argument
+ *   to 0. This is a way of saying "oops". The stack will stop
+ *   polling for datagrams, until there is a new call to
+ *   "picoquic_mark_datagram_ready"
+ * 
+ * - if the application does have data to send but the available
+ *   length indicated in the callback is too small, it should set the "length"
+ *   argument to 0, and the "is_active" argument to 0. The stack will try to
```

Comment:
shouldn't is_active be 1 here ?

Reply:
No. Setting "is_active" to 1 would start a hot loop.

Reply:
Good catch! It should be 1 indeed.